### PR TITLE
DA1469x i2c driver

### PR DESCRIPTION
**Pull request by blauret**
This PR is adding i2c driver for the Renesas DA1469x microcontroller family

### CODE REVIEW

**Comment by aaillet** [comment]
Hi @blauret,
Thanks for your first contribution !

Not talking about code itself yet, you have some changes to do to your commit tree in order to respect the project [contribution guidelines](https://docs.zephyrproject.org/latest/contribute/guidelines.html#commit-guidelines).

- You should remove any merge commit and use "fast forward" merge mechanism.
- As your adding a new driver/feature, you have to squash the "fix" commit as well as the "copyright" commit with the main "add driver" commit. Providing a fresh "v1" driver on the upstream repository.

While squashing commits, adding your changes to the main one, do not forget to add your signature to the commit alongside the existing one.

**Comment by blauret** [comment]
> Not talking about code itself yet, you have some changes to do to your commit tree in order to respect the project [contribution guidelines](https://docs.zephyrproject.org/latest/contribute/guidelines.html#commit-guidelines).
> 
> * You should rebase your work on main branch without any merge commit.
> * As your adding a new driver/feature, you have to squash the "fix" commit as well as the "copyright" commit with the main "add driver" commit. Providing a fresh "v1" driver on the upstream repository.
> 
> While squashing commits, adding your changes to the main one, do not forget to add your signature to the commit alongside the existing one.

Thanks for the feedback. I realized I'm not complying all the contributing guidelines. It's a learning process. 

Is there way to do that on this PR or do I need to build a new branch and a new PR?

**Comment by aaillet** [comment]
> Thanks for the feedback. I realized I'm not complying with all the contributing guidelines. It's a learning process.
> 
> Is there way to do that on this PR or do I need to build a new branch and a new PR?

By editing the branch on your fork, the PR will be updated accordingly !

Another thing you should do is describing the validation process you used to validate your driver as I think that no one will be able to test it on board.

Running Zephyr I2C tests suite on board could be a nice starting point.

**Comment by blauret** [comment]
> Another thing you should do is describing the validation process you used to validate your driver as I think that no one will be able to test it on board.
> 
> Running Zephyr I2C tests suite on board could be a nice starting point.

I'm not exactly sure what you refer to when you say I2C test suite. We built few tests onboard with external devices. Namely AT24 EEPROM, MCP9808. I did not think added the tests in the PR was a good idea.

**Comment by carlescufi** [comment]
@teburd  please review

File: boards/arm/da1469x_dk_pro/da1469x_dk_pro-pinctrl.dtsi [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2022 Renesas Electronics Corporation
+ * Copyright (c) 2022 Renesas Renesas Electronics Corporation and/or its affiliates
```
Comment (teburd) [inline boards/arm/da1469x_dk_pro/da1469x_dk_pro-pinctrl.dtsi:2]: Probably one Renesas here is enough :-)

**Review by teburd — DISMISSED** [review]
This looks fine, a few thoughts generally for suggested future improvements as you'll likely have people run into this pretty quickly.

The polling loops to do tx/rx in a spin lock may lead to long durations of delayed interrupts if the transfer length is long and/or the bus is clocked slow. I2C may be especially prone to this due to clock stretching. I'd recommend having a maximum length for polling transfers. Longer transfers could easily be redirect to the _async interrupt flow here and not prevent interrupts or scheduling.

I'd also consider putting a semaphore in place for the blocking transfer call. Inevitably someone will have two sensors on this bus and get a race trying to use it at the same. transfer_async can then use this with k_sem_take(sem, K_NO_WAIT) to either own the device or return -EBUSY.

File: boards/arm/da1469x_dk_pro/da1469x_dk_pro-pinctrl.dtsi (hunk 1 above)
Comment (blauret) [reply boards/arm/da1469x_dk_pro/da1469x_dk_pro-pinctrl.dtsi]: Thanks, good catch.

**Comment by blauret** [comment]
> This looks fine, a few thoughts generally for suggested future improvements as you'll likely have people run into this pretty quickly.
> 
> The polling loops to do tx/rx in a spin lock may lead to long durations of delayed interrupts if the transfer length is long and/or the bus is clocked slow. I2C may be especially prone to this due to clock stretching. I'd recommend having a maximum length for polling transfers. Longer transfers could easily be redirect to the _async interrupt flow here and not prevent interrupts or scheduling.
> 
> I'd also consider putting a semaphore in place for the blocking transfer call. Inevitably someone will have two sensors on this bus and get a race trying to use it at the same. transfer_async can then use this with k_sem_take(sem, K_NO_WAIT) to either own the device or return -EBUSY.

That's correct. We will need to improve that. I still would be happy to merge this one and look into improvement in a second step. 

@NiekIlmer, could you please have at the feedback from @teburd

**Review by teburd — DISMISSED** [review]

**Comment by blauret** [comment]
@galak , @aaillet , @MaureenHelm,

Anything we need to do to move this forward?

**Comment by blauret** [comment]
@andrzej-kaczmarek, Could you kindly review.

**Review by andrzej-kaczmarek — DISMISSED** [review]

**Review by sjanc — DISMISSED** [review]

**Comment by carlescufi** [comment]
@blauret can you please rebase to fix conflicts?

**Comment by blauret** [comment]
> @blauret can you please rebase to fix conflicts?

Done.

**Review by sjanc — DISMISSED** [review]

**Review by teburd — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**